ember-server/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala

```diff
@@ -101,12 +121,28 @@ private[server] object ServerHelpers {
               for {
                 socket <- connect.flatMap(upgradeSocket(_, tlsInfoOpt))
                 out <- Resource.liftF(
-                  Stream
-                    .eval(runApp(socket).attempt)
+                  (Stream(false) ++ Stream(true).repeat)
+                    .flatMap { isReused =>
+                      Stream
+                        .eval(
+                          runApp(socket, isReused).flatMap {
+                            case (req, resp) =>
+                              postProcessResponse(req, resp)
+                                .map(resp => (req, resp))
+                          }.attempt
+                        )
+                    }
                     .evalTap {
                       case Right((request, response)) => send(socket)(Some(request), response)
                       case Left(err) => send(socket)(None, onError(err))
                     }
+                    .takeWhile {
```

Comment:
if Error occurs or Connection hasClose is present on either request or response terminate the connection. 

Making sure we yield if we are still going to continue.